lib/sdk.js

```diff
@@ -0,0 +1,414 @@
+var blocks = require('block-stream2');
+var bz2 = require('unbzip2-stream');
+var createHash = require('sha.js');
+var fs = require('fs-extra-promise');
+var osenv = require('osenv');
+var path = require('path');
+var Progress = require('progress');
+var Promise = require('bluebird');
+var request = require('request');
+var spawn = require('child_process').spawn;
+var tar = require('tar-fs');
+var tmp = require('tmp');
+var Transform = require('stream').Transform;
+var zlib = require('zlib');
+
+var log = require('./log');
+
+var getPlatform = exports.getPlatform = () => {
+  switch (process.platform) {
+    case 'darwin':
+      return 'macos';
+    case 'linux':
+      return 'linux';
+    default:
+      throw new Error('Your platform is not yet supported for cross-compilation.');
+  }
+};
+
+var paths = {
+  sdk: path.join(osenv.home(), '.tessel/sdk'),
+  rustlib: path.join(osenv.home(), '.tessel/rust'),
+};
+
+var sdkPath = {
+  'macos': 'https://builds.tessel.io/t2/sdk/t2-sdk-macos-x86_64.tar.bz2',
+  'linux': 'https://builds.tessel.io/t2/sdk/t2-sdk-linux-x86_64.tar.bz2',
+};
+
+var rustlibUrl = 'https://builds.tessel.io/t2/sdk/t2-rustlib-VERSION.tar.gz';
```

Comment:
Use const casing, eg. `RUST_LIB_TGZ_URL` (eg. see `RESTORE_TGZ_URL` in update-fetch.js)